src/dev/impl/DevToys/ViewModels/Tools/Graphic/ImageConverter/ImageConversionWorkItem.cs

```diff
@@ -0,0 +1,168 @@
+
+            FileName = file.Name;
+            FilePath = file.Path;
+
+            using (IRandomAccessStream stream = Task.Run(async () => await file.OpenAsync(FileAccessMode.Read)).Result)
+            {
+                BitmapDecoder decoder = Task.Run(async () => await BitmapDecoder.CreateAsync(stream)).Result;
+                Bitmap = Task.Run(async () => await decoder.GetSoftwareBitmapAsync()).Result;
+            }
+
+            DeleteCommand = new RelayCommand(ExecuteDeleteCommand);
+            SaveCommand = new AsyncRelayCommand<ImageConverterToolViewModel>(ExecuteSaveCommandAsync);
+            ShowErrorMessageCommand = new AsyncRelayCommand(ExecuteShowErrorMessageCommandAsync);
+
+            ComputePropertiesAsync(file).Forget();
+        }
+
+        #region DeleteCommand
+
+        public IRelayCommand DeleteCommand { get; }
+
+        private void ExecuteDeleteCommand()
+        {
+            DeleteItemRequested?.Invoke(this, EventArgs.Empty);
+        }
+
+        #endregion
+
+        #region SaveCommand
+
+        public IAsyncRelayCommand SaveCommand { get; }
+
+        private async Task ExecuteSaveCommandAsync(ImageConverterToolViewModel viewModel)
+        {
+            string? fileExtension = ImageHelper.GetExtension(viewModel.ConvertedFormat);
+
+            var savePicker = new FileSavePicker
+            {
+                SuggestedStartLocation = PickerLocationId.ComputerFolder
+            };
+
+            savePicker.FileTypeChoices.Add(
+                fileExtension.Replace(".", string.Empty).ToUpperInvariant(),
+                new List<string>() { fileExtension!.ToLowerInvariant() });
+
+            StorageFile? newFile = await savePicker.PickSaveFileAsync();
+
+            if (newFile is not null)
+            {
+                using (IRandomAccessStream outputStream = await newFile.OpenAsync(FileAccessMode.ReadWrite))
+                {
+                    BitmapEncoder? encoder = await ImageHelper.GetEncoderAsync(viewModel.ConvertedFormat, outputStream);
+
+                    encoder.SetSoftwareBitmap(Bitmap);
+                    await encoder.FlushAsync();
+                }
+
+                DeleteCommand.Execute(this);
+            }
+        }
+
+        #endregion
+
+        #region ShowErrorMessageCommand
+
+        public IAsyncRelayCommand ShowErrorMessageCommand { get; }
+
+        private async Task ExecuteShowErrorMessageCommandAsync()
+        {
+            var dialog = new ContentDialog
+            {
+                Title = Strings.DetailsTitle,
+                CloseButtonText = Strings.OK,
+                DefaultButton = ContentDialogButton.Close,
+                Content = ErrorMessage!
+            };
+
+            await dialog.ShowAsync();
+        }
+
+        #endregion
+
+        private async Task ComputePropertiesAsync(StorageFile file)
+        {
+            BasicProperties fileProperties = await file.GetBasicPropertiesAsync();
```

Comment:
Please add at the beginning of this method `await TaskScheduler.Default`. This will make sure we go off the UI Thread.

It's possible we're missing this in PNG / JPEG Compressor too (oops).